src/Form/DataTransformer/ModelsToArrayTransformer.php

```diff
@@ -57,23 +57,25 @@ public function __construct(ModelManagerInterface $modelManager, string $class)
      *
      * @phpstan-param T[]|null $value
      */
-    public function transform($value)
+    public function transform($value): array
     {
         if (null === $value) {
             return [];
         }
 
         $array = [];
         foreach ($value as $model) {
-            $id = implode(AdapterInterface::ID_SEPARATOR, $this->getIdentifierValues($model));
+            \assert($model instanceof $this->class);
```

Comment:
It's weird that static analysis need this, since we have
- `@phpstan-param class-string<T> $class`
- `@phpstan-param T[]|null $value`

Reply:
Fixed